modules/core/keeper/keeper.go

```diff
@@ -62,23 +64,37 @@ func NewKeeper(
 	clientKeeper := clientkeeper.NewKeeper(cdc, key, paramSpace, stakingKeeper, upgradeKeeper)
 	connectionKeeper := connectionkeeper.NewKeeper(cdc, key, paramSpace, &clientKeeper)
 	portKeeper := portkeeper.NewKeeper(scopedKeeper)
-	channelKeeper := channelkeeper.NewKeeper(cdc, key, &clientKeeper, &connectionKeeper, &portKeeper, scopedKeeper)
+	channelKeeper := channelkeeper.NewKeeper(cdc, key, clientKeeper, &connectionKeeper, &portKeeper, scopedKeeper)
 
-	return &Keeper{
+	keeper := &Keeper{
 		cdc:              cdc,
-		ClientKeeper:     &clientKeeper,
+		ClientKeeper:     clientKeeper,
 		ConnectionKeeper: connectionKeeper,
 		ChannelKeeper:    channelKeeper,
 		PortKeeper:       &portKeeper,
 		authority:        authority,
 	}
+
+	keeper.SetConsensusHost(ibctm.NewConsensusHost(stakingKeeper))
+
+	return keeper
 }
 
 // Codec returns the IBC module codec.
 func (k Keeper) Codec() codec.BinaryCodec {
 	return k.cdc
 }
 
+// SetConsensusHost sets a custom ConsensusHost for self client state and consensus state validation.
+func (k *Keeper) SetConsensusHost(consensusHost clienttypes.ConsensusHost) {
```

Comment:
the idea was to expose this one and have users/devs user this rather than having to make two calls. 

This would hopefully only exist in this release line and v9 could do away with this